### PR TITLE
Ability to create Jenkins worker AMIs via packer.

### PR DESCRIPTION
**Pull request by benpatterson**
@jzoldak @clytwynec 

In progress but wanted to post what I have so far. Please take a look?

### CODE REVIEW

**Comment by benpatterson** [comment]
@jzoldak @clytwynec 

I've rebased and added the bit where we're testing by running the test_build_server role.

**Comment by clytwynec** [comment]
@benpatterson  Can you describe the actual process that we'd have to go through to get a new ami set up with these changes?

**Comment by benpatterson** [comment]
@clytwynec @jzoldak 

A new AMI would be created as follows:
- via shell (or shell script), 
  *\* set the needed environment variables (AMI id, AWS keys & secrets, github oauth ID)
  *\* cd into the util/packer directory
  *\* run 

```
packer build jenkins_worker.json
```
- If successful, a private AMI will be created with a timestamp
- we would then manually change the ami-id assigned to jenkins workers, and the new ami will begin to be used when the others expire, or we force-down the older workers.

Ultimately the AMI creation would be executed by the public jenkins instance in a job that is not visible to the public. We can re-use the AWS keys that are used for turning on/off workers.

If you'd rather, I can wiki-fy this as well.

cc @feanil @hkim823

File: util/packer/jenkins_worker.json [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  }, {
+    "type": "file",
+    "source": "../../../configuration/playbooks",
+    "destination": "{{user `playbook_remote_dir`}}"
+  }, {
+    "type": "file",
+    "source": "../../../configuration/requirements.txt",
+    "destination": "{{user `playbook_remote_dir`}}/requirements.txt"
+  },
+  {
```
Comment (clytwynec) [inline util/packer/jenkins_worker.json]: I think this has an extra new line between the `,` and  `{`

Comment (benpatterson) [inline util/packer/jenkins_worker.json]: Quite right. Fixed.

**Comment by clytwynec** [comment]
:+1:

**Comment by clytwynec** [comment]
https://github.com/edx/configuration/pull/1443 Is merged ...

**Comment by benpatterson** [comment]
squashed commits and merging

**Pull request merged**